.github/workflows/scripts/check_tracking_issue_closure

```diff
@@ -0,0 +1,279 @@
+
+# Error handler.
+on_error() {
+	echo "Error: An error was encountered during execution." >&2
+	exit 1
+}
+
+# Exit handler.
+on_exit() {
+	echo "Script execution completed."
+	return 0
+}
+
+# Performs a GitHub API request.
+#
+# $1 - HTTP method (GET or POST)
+# $2 - API endpoint
+# $3 - data for POST requests
+github_api() {
+	local method="$1"
+	local endpoint="$2"
+	local data="$3"
+
+	# Initialize an array to hold curl headers:
+	local headers=()
+
+	# If GITHUB_TOKEN is set, add the Authorization header:
+	if [ -n "${github_token}" ]; then
+		headers+=("-H" "Authorization: token ${github_token}")
+	fi
+
+	# Determine the HTTP method and construct the curl command accordingly...
+	case "${method}" in
+		GET)
+			curl -s "${headers[@]}" "${github_api_url}${endpoint}"
+			;;
+		POST)
+			# For POST requests, always set the Content-Type header:
+			headers+=("-H" "Content-Type: application/json")
+
+			# If data is provided, include it in the request:
+			if [ -n "${data}" ]; then
+				curl -s -X POST "${headers[@]}" -d "${data}" "${github_api_url}${endpoint}"
+			else
+				# Handle cases where POST data is required but not provided:
+				echo "ERROR: POST request requires data." >&2
+				return 1
+			fi
+			;;
+		*)
+			echo "ERROR: Invalid HTTP method: ${method}." >&2
+			return 1
+			;;
+	esac
+}
+
+# Get date in ISO 8601 format for N days ago.
+#
+# $1 - Number of days ago
+get_date_n_days_ago() {
+	local days="$1"
+
+	# Check if we're on macOS or Linux:
+	if [[ "$(uname)" == "Darwin" ]]; then
+		# macOS date command:
+		date -u -v-"${days}"d "+%Y-%m-%dT%H:%M:%SZ"
+	else
+		# Linux date command:
+		date -u -d "${days} days ago" "+%Y-%m-%dT%H:%M:%SZ"
+	fi
+}
+
+# Check if a PR has already been commented on by the bot regarding tracking issue closure.
+#
+# $1 - PR number
+has_bot_comment() {
+	local pr_number="$1"
+	local response
+
+	# Get all comments on the PR:
+	response=$(github_api "GET" "/repos/${repo_owner}/${repo_name}/issues/${pr_number}/comments")
+
+	# Check if any comment contains our unique identifier:
+	if echo "$response" | jq -r '.[] | .body' | grep -q "${comment_identifier}"; then
+		return 0
+	else
+		return 1
+	fi
+}
+
+# Check if an issue has the "Tracking Issue" label.
+#
+# $1 - Issue number
+is_tracking_issue() {
+	local issue_number="$1"
+	local response
+
+	# Get the issue:
+	response=$(github_api "GET" "/repos/${repo_owner}/${repo_name}/issues/${issue_number}")
+
+	# Check if the issue exists and has the "Tracking Issue" label:
+	if echo "$response" | jq -r '.labels[].name' | grep -q "Tracking Issue"; then
+		return 0
+	else
+		return 1
+	fi
+}
+
+# Post a comment on a PR.
+#
+# $1 - PR number
+post_comment() {
+	local pr_number="$1"
+	local comment_body
+	local json_payload
+
+	comment_body="${comment_identifier}
+:warning: **Tracking Issue Closure Warning** :warning:
+
+I noticed your PR description contains closing keywords (\"Resolves\", \"Closes\", or \"Fixes\") referencing a \"Tracking Issue\".
```

Comment:
May be good to link to the GitHub docs here, as well: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

Reply:
Or maybe in the "Why this matters" section.